src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java

```diff
@@ -576,6 +672,54 @@ public Result onBuildCompleted(Participation participation) {
         return result.getFeedbacks();
     }
 
+    /**
+     * Converts build result details into feedback and stores it in the result object
+     * @param result the result for which the feedback should be added
+     * @param failedJobs the failedJobs list of the requestBody
+     *
+     * @return a list of feedbacks itemsstored in a result
+     */
+    public List<Feedback> addFeedbackToResultNew(Result result, List<Object> failedJobs) {
+        if(failedJobs == null) {
+            return null;
+        }
+
+        try {
+            List<Map<String, Object>> castedfailedJobs = (List<Map<String, Object>>) (Object) failedJobs; // TODO: check if this works correctly
+
+            for (Map<String, Object> failedJob : castedfailedJobs) {
+                for (Map<String, Object> failedTest : (List<Map<String, Object>>) (Object) failedJob.get("failedTests")) {
+                    result.setHasFeedback(true); // TODO: check if setting this multiple times is a performance issue
```

Comment:
this should not be a performance problem. However I still suggest to keep it outside the loop and set it if at least one failedTests object exists

Reply:
I've moved it into the outer for-loop, so that it will only be set if at least one failedTest-object exists.